# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/FuzzySecurityInfo.cpp
# Commit: 7f0065ff7b82
# Full Hash: 7f0065ff7b8200c193d0986f1ce2fa3f30d4a99d
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-22 15:53:06
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Fix fuzzing build, also we cannot set SecurityInfo during read r=necko-reviewers,kershaw
#   
#   Before this bug TLS handshake was only driven by forcing writes. SecurtyInfo was set during a write code path. That is not anymore true and the TLS handshake can be driven by reading from a socket. That causes an issue where the SecurtyInfo was not set in case a TLS handshake fails. This bug added the setting of the SecurtyInfo to the read code path, but that causes problems when the transaction is closed due to corrupted response.
#   This patch fixes this by moving the setting of SecurtyInfo to Close() function.
#   
# ==============================================================================

diff -r 4150724f355f -r 7f0065ff7b82 netwerk/base/FuzzySecurityInfo.cpp
--- a/netwerk/base/FuzzySecurityInfo.cpp	Wed Sep 22 09:19:32 2021 +0000
+++ b/netwerk/base/FuzzySecurityInfo.cpp	Wed Sep 22 09:19:32 2021 +0000
@@ -369,5 +369,10 @@
   return NS_OK;
 }
 
+NS_IMETHODIMP FuzzySecurityInfo::SetHandshakeCallbackListener(
+    nsITlsHandshakeCallbackListener* callback) {
+  return NS_OK;
+}
+
 }  // namespace net
 }  // namespace mozilla